grades/models.py

```diff
@@ -79,3 +90,60 @@ def __str__(self):
             user=self.final_grade.user,
             course_id=self.final_grade.course_run.edx_course_key
         )
+
+
+class FinalGradeRunInfo(TimestampedModel):
+    """
+    Additional information for the course run related to the final grades
+    """
+    course_run = models.OneToOneField(CourseRun, null=False)
+    status = models.CharField(
+        null=False,
+        choices=[(status, status) for status in FinalGradeStatus.ALL_STATUSES],
+        default=FinalGradeStatus.PENDING,
+        max_length=30,
+    )
+
+    @classmethod
+    def is_course_run_complete(cls, course_run):
+        """
+        Returns True if there is an entry with status 'complete'
+        """
+        return cls.objects.filter(course_run=course_run, status=FinalGradeStatus.COMPLETE).exists()
+
+    @classmethod
+    def is_course_run_pending(cls, course_run):
+        """
+        Returns True if there is an entry with status 'complete'
+        """
+        return cls.objects.filter(course_run=course_run, status=FinalGradeStatus.PENDING).exists()
+
+    @classmethod
+    def complete_course_run(cls, course_run):
+        """
+        Sets the status for the course_run to complete
+        """
+        course_run_grade_status, _ = cls.objects.get_or_create(course_run=course_run)
+        if course_run_grade_status.status != FinalGradeStatus.COMPLETE:
+            course_run_grade_status.status = FinalGradeStatus.COMPLETE
+            course_run_grade_status.save()
+        return course_run
+
+    @classmethod
+    def create_pending_course_run(cls, course_run):
+        """
+        Creates an entry with status pending
+        """
+        course_fg_info, _ = cls.objects.get_or_create(course_run=course_run)
+        return course_fg_info
```

Comment:
we already know this model deals with course runs, and these methods all accept a `course_run` argument. including `course_run` in the method name is kinda redundant. some renaming suggestions:

`is_course_run_complete` -> `is_complete`
`is_course_run_pending` -> `is_pending`
`complete_course_run` -> `set_to_complete` (complete is an adjective and a verb, so 'set_to' is more helpful)
`create_pending_course_run` -> `create_pending`